themes/bootstrap3/js/check_save_statuses.js

```diff
@@ -1,58 +1,117 @@
-/*global htmlEncode, userIsLoggedIn, VuFind */
+/*global htmlEncode, userIsLoggedIn, Hunt, VuFind */
 /*exported checkSaveStatuses, checkSaveStatusesCallback */
 
-function checkSaveStatuses(_container) {
-  if (!userIsLoggedIn) {
+function displaySaveStatus(itemLists, $item) {
+  $item.removeClass('.ajax-pending');
+  if (itemLists.length > 0) {
+    var html = '<ul>' + itemLists.map(function convertToLi(l) {
+      return '<li><a href="' + l.list_url + '">' + htmlEncode(l.list_title) + '</a></li>';
+    }).join('') + '</ul>';
+    $item.find('.savedLists').html($item.find('.savedLists strong')[0].outerHTML + html).addClass('loaded');
+  }
+}
+
+function saveStatusFail(response, textStatus) {
+  $('.savedLists.ajax-pending').empty();
+  if (textStatus === 'abort' || typeof response.responseJSON === 'undefined') {
     return;
   }
-  var container = _container || $('body');
+  // display the error message on each of the ajax status place holder
+  $('.savedLists.ajax-pending').addClass('alert-danger').append(response.responseJSON.data);
+}
 
-  var elements = {};
-  var data = $.map(container.find('.result,.record'), function checkSaveRecordMap(record) {
-    if ($(record).find('.hiddenId').length === 0 || $(record).find('.hiddenSource').length === 0) {
-      return null;
+var saveStatusObjs = [];
+var saveStatusEls = {};
+var saveStatusTimer = null;
+var saveStatusDelay = 200;
+var saveStatusRunning = false;
+
+function runSaveAjaxForQueue() {
+  // Only run one save status AJAX request at a time:
+  if (saveStatusRunning) {
```

Comment:
I see that you've added saveStatusRunning, but it looks like you are not clearing it. I think it needs to get set to false in the .done and .fail callbacks of the AJAX routine or else only the first batch will load and then we'll go into an infinite wait loop.